backend/gateway/mux/mux.go

```diff
@@ -62,14 +69,53 @@ func (a *assetHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	// Serve!
 	if f, err := a.fileSystem.Open(r.URL.Path); err != nil {
+		// If not a known static asset and an asset provider is configured, try streaming from the configured provider.
+		if a.assetCfg != nil {
+			if a.assetCfg.Provider != nil && strings.HasPrefix(r.URL.Path, "/static") {
+				// We attatch this header simply for observibility purposes.
+				// Otherwise its difficult to know if the assets are being served from the configured provider.
+				w.Header().Set("x-clutch-asset-passthrough", "true")
+
+				// TODO: handle errors
+				asset, _ := a.assetProviderHandler(r.URL.Path)
+				_, _ = io.Copy(w, asset)
+				return
+			}
+		}
+
 		// If not a known static asset serve the SPA.
 		r.URL.Path = "/"
 	} else {
 		_ = f.Close()
 	}
+
 	a.fileServer.ServeHTTP(w, r)
 }
 
+func (a *assetHandler) assetProviderHandler(urlPath string) (io.Reader, error) {
+	switch a.assetCfg.Provider.(type) {
+	case *gatewayv1.Assets_S3:
+		aws, ok := service.Registry[awsservice.Name]
+		if !ok {
+			return nil, fmt.Errorf("The AWS service must be configured to use the asset s3 provider.")
+		}
+
+		awsClient, ok := aws.(awsservice.Client)
+		if !ok {
+			return nil, fmt.Errorf("Unable to aquire the aws client")
+		}
+
+		return awsClient.S3StreamingGet(
+			context.Background(),
+			a.assetCfg.GetS3().Region,
+			a.assetCfg.GetS3().Bucket,
+			path.Join(a.assetCfg.GetS3().Key, strings.TrimPrefix(urlPath, "/static")),
+		)
+	default:
+		return nil, fmt.Errorf("No asset provider is configured")
```

Comment:
i think this error is actually 
```suggestion
		return nil, fmt.Errorf("configured asset provider has not been implemented")
```